arm/Microsoft.ContainerService/managedClusters/agentPools/deploy.bicep

```
@@ -5,17 +5,503 @@ param managedClusterName string
+
+@description('Optional. Sysctl setting net.ipv4.tcp_max_tw_buckets.')
+@minValue(8000)
+@maxValue(1440000)
+param netIpv4TcpMaxTwBuckets int = 8000
+
+@description('Optional. Sysctl setting net.ipv4.tcp_tw_reuse.')
+@allowed([
+  true
+  false
+])
+param netIpv4TcpTwReuse bool = false
+
+@description('Optional. Sysctl setting net.netfilter.nf_conntrack_buckets.')
+@minValue(65536)
+@maxValue(147456)
+param netNetfilterNfConntrackBuckets int = 65536
+
+@description('Optional. Sysctl setting net.netfilter.nf_conntrack_max.')
+@minValue(131072)
+@maxValue(589824)
+param netNetfilterNfConntrackMax int = 131072
+
+@description('Optional. Sysctl setting vm.max_map_count.')
+@minValue(65530)
+@maxValue(262144)
+param vmMaxMapCount int = 65530
+
+@description('Optional. Sysctl setting vm.swappiness.')
+@minValue(0)
+@maxValue(100)
+param vmSwappiness int = 0
+
+@description('Optional. Sysctl setting vm.vfs_cache_pressure.')
+@minValue(0)
+@maxValue(100)
+param vmVfsCachePressure int = 0
+
+@description('Optional. See Transparent Hugepages (https://www.kernel.org/doc/html/latest/admin-guide/mm/transhuge.html#admin-guide-transhuge)')
+@allowed([
+  'always'
+  'defer'
+  'defer+madvise'
+  'madvise'
+  'never'
+])
+param transparentHugePageDefrag string = 'madvise'
+
+@description('Optional. See Transparent Hugepages (https://www.kernel.org/doc/html/latest/admin-guide/mm/transhuge.html#admin-guide-transhuge)')
+@allowed([
+  'always'
+  'madvise'
+  'never'
+])
+param transparentHugePageEnabled string = 'always'
+
+@description('Optional. The maximum number of nodes for auto-scaling')
+param maxCount int = -1
+
+@description('Optional. The maximum number of pods that can run on a node.')
+param maxPods int = -1
+
+@description('Optional. The minimum number of nodes for auto-scaling')
+param minCount int = -1
+
+@description('Optional. A cluster must have at least one "System" Agent Pool at all times. For additional information on agent pool restrictions and best practices, see: /azure/aks/use-system-pools')
+param mode string = ''
+
+@description('Optional. The node labels to be persisted across all nodes in agent pool.')
+param nodeLabels object = {}
+
+@description('Optional. ResourceId of the node PublicIPPrefix')
+param nodePublicIpPrefixId string = ''
+
+@description('Optional. The taints added to new nodes during node pool create and scale. For example, key=value:NoSchedule.	')
+param nodeTaints array = []
+
+@description('Optional. 	As a best practice, you should upgrade all node pools in an AKS cluster to the same Kubernetes version. The node pool version must have the same major version as the control plane. The node pool minor version must be within two minor versions of the control plane version. The node pool version cannot be greater than the control plane version. For more information see upgrading a node pool (https://docs.microsoft.com/en-us/azure/aks/use-multiple-node-pools#upgrade-a-node-pool).')
```

Remove redundant spaces
```suggestion
@description('Optional. As a best practice, you should upgrade all node pools in an AKS cluster to the same Kubernetes version. The node pool version must have the same major version as the control plane. The node pool minor version must be within two minor versions of the control plane version. The node pool version cannot be greater than the control plane version. For more information see upgrading a node pool (https://docs.microsoft.com/en-us/azure/aks/use-multiple-node-pools#upgrade-a-node-pool).')
```

---

Removed extra spaces.

---

are you sure?

---

Was in the middle up making changes, latest code has been uploaded.